spec.bs

```diff
@@ -18,3 +18,85 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=dictionary=] with the following:
+
+* A <dfn for=advertisement>renderURL</dfn>.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data.
+
+<xmp class="idl">
+dictionary AuctionAd {
+  required USVString renderUrl;
+  any metadata;
+};
+</xmp>
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=dictionary=] with the following:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.
+
+* An <dfn for="advertisement interest group">owner origin</dfn>, a [=valid url string=] whose [=url/origin=] will be used as the owner of the interest group.
+
+* A <dfn for="advertisement interest group">name</dfn>, a [=string=].
+ 
+* A <dfn for="advertisement interest group">priority</dfn>, a {{double}}, initially 0.
+ 
+* An <dfn for="advertisement interest group">enableBiddingSignalsPrioritization</dfn>, a [=boolean=], initially false.
+ 
+* An optional <dfn for="advertisement interest group">priorityVector</dfn>, A [=map=] of {{double}}.
+ 
+* An optional <dfn for="advertisement interest group">prioritySignalsOverrides</dfn>, A [=map=] of {{double}}.
+ 
+* An <dfn for="advertisement interest group">execution_mode</dfn>, a {{WorkletExecutionMode}}.
+ 
+* An optional <dfn for="advertisement interest group">biddingLogicURL</dfn>, a [=url=].
+
+* An optional <dfn for="advertisement interest group">biddingWasmHelperUrl</dfn>, a [=url=].
+ 
+* An optional <dfn for="advertisement interest group">dailyUpdateUrl</dfn>, a [=url=].
+
+* An optional <dfn for="advertisement interest group">trustedBiddingSignalsUrl</dfn>, a [=url=].
+
+* A <dfn for="advertisement interest group">trustedBiddingSignalsKeys</dfn>, a [=list=] of [=string=].
+ 
+* An optional <dfn for="advertisement interest group">userBiddingSignals</dfn>, a [=string=].
+ 
+* An <dfn for="advertisement interest group">ads</dfn>, a [=list=] of [=advertisement=].
+
+* An <dfn for="advertisement interest group">adComponents</dfn>, a [=list=] of [=advertisement=].
+
+<xmp class="idl">

```

Comment:
should we be putting web idl here?  I thought this would go next to the JS API sections where it's used

Reply:
I was just adding the idl so that we could reference between infra and idl definitions.  Everything can be moved around later.  Jeffrey suggests we should start by adding the functions, and then we only need to define types for parameter values and results.